openhcl/virt_mshv_vtl/src/processor/hardware_cvm/mod.rs

```diff
@@ -900,6 +907,102 @@ impl<T, B: HardwareIsolatedBacking> hv1_hypercall::ModifyVtlProtectionMask
     }
 }
 
+impl<T, B: HardwareIsolatedBacking> hv1_hypercall::TranslateVirtualAddressX64
+    for UhHypercallHandler<'_, '_, T, B>
+{
+    fn translate_virtual_address(
+        &mut self,
+        partition_id: u64,
+        vp_index: u32,
+        control_flags: hvdef::hypercall::TranslateGvaControlFlagsX64,
+        gva_page: u64,
```

Comment:
Side note: on ARM, if the guest configures a 64KB page size, does this `gva_page` parameter refer to a 64KB page or a 4KB page?

Of course on x86 we only have 4KB paging, so this is irrelevant. But I want to understand the hypervisor's mental model here of what these input parameters mean.

Reply:
Getting confirmation on the answer to this.

Reply:
The hypervisor always uses 4KB granularity at stage 2, even if the guest uses 64KB in stage 1. So gva_page refers to a  4KB page.

Reply:
Yeah, that's what I figured, because anything else would be impossible to use; the caller would have to know the granule size in order to call it.

But at the same time, it doesn't really make sense, because this function is all about stage 1 translation, not stage 2.

(The hypercall should have taken a full GVA and not GVA/4096. But that ship has sailed.)